src/unit/test_object.c

```diff
@@ -48,3 +48,31 @@ int test_object_with_key(int argc, char **argv, int flags) {
     decrRefCount(valkey);
     return 0;
 }
+
+int test_embedded_string_with_key(int argc, char **argv, int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    /* key of length 32 - type 8 */
+    sds key = sdsnew("k:123456789012345678901234567890");
+
+    /* value of length 7 should be embedded */
+    const char *short_value = "v:12345";
+    robj *short_val_obj = createStringObject(short_value, strlen(short_value));
+
+    /* value of length 8 should be raw, because the total size is over 64 */
+    const char *longer_value = "v:123456";
+    robj *longer_val_obj = createStringObject(longer_value, strlen(longer_value));
+
+    robj *embstr_obj = objectSetKeyAndExpire(short_val_obj, key, -1);
+    TEST_ASSERT(embstr_obj->encoding == OBJ_ENCODING_EMBSTR);
+
+    robj *raw_obj = objectSetKeyAndExpire(longer_val_obj, key, -1);
+    TEST_ASSERT(raw_obj->encoding == OBJ_ENCODING_RAW);
```

Comment:
This test case fails in the daily runs for 32-bit:

* https://github.com/valkey-io/valkey/actions/runs/15125828365/job/42517576263#step:10:349
* https://github.com/valkey-io/valkey/actions/runs/15101473588/job/42442959926#step:10:344

@poiuj Can you look into it?

Reply:
A pointer is 32 bits so the robj struct is smaller.